dev-docs/source/Gatekeeping.rst

```diff
@@ -0,0 +1,156 @@
+.. _Gatekeeping:
+
+==========================
+Gatekeeping a Pull Request
+==========================
+
+The final step in our :ref:`development workflow <GitWorkflow>` after a :ref:`pull request has been reviewed <ReviewingAPullRequest>` is gatekeeping. This is second review stage where a senior developer / code owner does a final check of the work before merging it to the ``main`` branch.
+
+Purpose
+=======
+
+This document offers some guidance for gatekeepers as to what to look for and some quirks of Mantid to be aware of that the first reviewer may have missed. There are 3 main purposes to gatekeeping:
+
+* **Audit:** A detailed code review is generally not required as long as the first stage reviewer has done a thorough job - one of the main purposes of the gatekeeping role is to ensure that this is the case, i.e. that the review process has been followed correctly and all of the appropriate checks have been made.
+* **Code quality:** The other main purpose is to ensure all code changes have been seen by an experienced developer. We very much encourage new/junior team members to take part in first-stage reviews, so having this second stage ensures we have a chance to catch things that a newer team member may not be aware of.
+* **Knowledge sharing:** An additional benefit of the process is the opportunity for more experienced developers to offer advice and spot learning and development opportunities for individuals and/or things the whole team could benefit from. Depending on your capacity and interest in this do feel free to offer tips and/or feed into team processes or training materials as you think appropriate.
+
+
+With great power comes great responsibility
+===========================================
+
+*Be very careful not to push to* ``main``
+
+* When you are made a gatekeeper on the project you will be given **full, unprotected access to the** ``main`` **branch**. This means you lose the normal protection that stops you being able to push directly to the ``main`` branch - you therefore need to be careful at all times that you do not accidentally push to ``main``, e.g. when you are using Git in your terminal or Git GUI.
```

Comment:
I don't think this is entirely correct. Project **owners** have unprotected access (although they are still protected against rewritng history with `--force`). I believe gatekeekers can't push to `main` without an approved PR with passing status checks?

Reply:
I wasn't sure actually and I couldn't find any documentation to confirm one way or the other

Reply:
Ok, after a lot of digging:

Gatekeepers have the [admin role](https://github.com/mantidproject/mantid/settings/access), and [this page](https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches) says "By default, the restrictions of a branch protection rule don't apply to people with admin permissions to the repository". 

In the branch protection settings:
- `Do not allow bypassing the above settings ` is _not_ enabled so we _do_ allow bypassing protection for admins and others  with "bypass branch protection" permission.
- We have `Restrict who can push to matching branches` enabled and gatekeepers _are_ included, so they can push directly to `main`.
- We have `Allow force pushes` and `Allow deletions` disabled though so no one can force push (phew!). 

We might want to reconsider the roles. It seems unnecessary that gatekeepers have full admin access.